app/Content/src/Http/Controllers/BlogPostController.php

```diff
@@ -0,0 +1,132 @@
+    #[BodyParam('content', 'string', required: true, example: "Lorem ipsum dolor sit amet, consectetur adipiscing ...")]
+    #[BodyParam('tags', 'array', required: true, example: "[1,2,3]")]
+    public function store(CreateBlogPostRequest $request): BlogPostsResource
+    {
+        $blogPost = BlogPost::create([
+            'title' => $request->title,
+            'slug' => Str::slug($request->title),
+            'excerpt' => $request->excerpt,
+            'content' => $request->content,
+            'status' => BlogPostStatus::IN_REVIEW,
+            'author_id' => Author::where('user_id', $request->user()->id)->value('id')
+        ]);
+
+        $blogPost->tags()->sync($request->tags);
+
+        return new BlogPostsResource($blogPost);
+    }
+
+    #[Endpoint(title: 'Blog post', description: 'This endpoint returns a single blog post')]
+    #[Group('Content')]
+    public function show(BlogPost $blogPost, BlogPostPermissionsRequest $request): BlogPostsResource
+    {
+        $blogPost->load('author', 'tags');
+
+        return new BlogPostsResource($blogPost);
+    }
+
+    #[Endpoint(title: 'Update Blog post', description: 'This endpoint will update a single blog post')]
+    #[Group('Content')]
+    #[BodyParam('title', 'string', required: false, example: "Example Blog Post Title ")]
+    #[BodyParam('slug', 'string', required: false, example: "example-blog-post-title ")]
+    #[BodyParam('excerpt', 'string', required: false, example: "This is test blogpost example")]
+    #[BodyParam('content', 'string', required: false, example: "Lorem ipsum dolor sit amet, consectetur adipiscing ...")]
+    #[BodyParam('tags', 'array', required: false, example: "[1,2,3]")]
+    public function update(UpdateBlogPostRequest $request, BlogPost $blogPost): BlogPostsResource
+    {
+        $data = $request->validated();
+
+        $blogPostData = (new Collection($data))->except('tags')->all();
+
+        $blogPost->update($blogPostData);
+
+        if ($request->has('tags')) {
+            $blogPost->tags()->sync($request->tags);
+        }
+
+        return new BlogPostsResource($blogPost);
+
+    }
+
+    #[Endpoint(title: 'Delete Blog posts', description: 'This endpoint deletes blog post')]
+    #[Group('Content')]
+    public function destroy(BlogPost $blogPost, BlogPostPermissionsRequest $request): \Illuminate\Http\Response
+    {
+        $blogPost->delete();
+
+        return response()->noContent();
+    }
+
+    #[Endpoint(title: 'Publish/Unpublish Blog posts', description: 'This endpoint publish or unpublish blog post')]
+    #[Group('Content')]
+    public function changeStatus(BlogPost $blogPost, BlogPostPermissionsRequest $request): \Illuminate\Http\Response
+    {
+        if ($blogPost->status == BlogPostStatus::PUBLISHED) {
+            $blogPost->update(['status' => BlogPostStatus::DRAFT, 'publish_date' => null]);
+        } else {
+            $blogPost->update(['status' => BlogPostStatus::PUBLISHED, 'publish_date' => now()]);
```

Comment:
It may make sense to pass the publish date as a parameter so that it can be scheduled and shown in the future.

Reply:
Did you make any changes here? What do you think about this one?

Additionally, I am thinking that you could get a status from the request and update it that way but also ask for a publish date in the process if the status is `published`.